sparkly/session.py

```diff
@@ -89,6 +109,15 @@ def __init__(self, additional_options=None):
         options = {
             'spark.sql.catalogImplementation': 'hive',
         }
+        if self.app_id_template:
```

Comment:
What do you think about introducing a property/method like `sparkly_app_id`?
It might bring additional flexibility:
```
class MySession(SparklySession):
     @property
     def sparkly_app_id(self):
          return f'{any}{variable}'
```

Reply:
Technically, this could be achieved by supplying in the child class an override for `app_id_template`, which also has the benefit of being able to access the 4 automatically-provided variables:

```
@property
def app_id_template(self):
  return f'{any}{variable}{visible}{in}{this}{scope}-{{session_time}}'
```

I foresee very little subclass implementations for this app id in reality, and I didn't want them to have to figure out process-uid and session-time themselves, which is why I stuck it as a template; it makes it a one-liner to change how the string is formed, or say to give a prefix according to the name of the service (or something about the context of the session by using syntax above.  Of course I could do as you suggest with something like this:

```
class SparklySession:

  app_id_template = 'sparkly-{process_uid}-{session_time}'    # or a @property

  @property
  def sparkly_app_id(self):
    return app_id_template.format(...)
```

But then it seems unecessarily complex ("which one do I override?").